packages/core/src/StateMachine.ts

```diff
@@ -1,6 +1,13 @@
-import { AnyStateMachine, InvokeActionObject, Spawner, StateFrom } from '.';
+import {
```

Comment:
With this change, I feel that `StateMachine` should implement a `Behavior` interface.

Reply:
Needs to be done when `v5/simplify-algorithm` is merged into this branch because of the state/event changes

Reply:
since it has landed it could be done now, right?

Reply:
I created a draft PR in which I try to implement `Behavior` in the `StateMachine`: https://github.com/statelyai/xstate/pull/3696

There are some compatibility issues there - mainly around the `actorContext` argument. Currently `StateMachine` assumes that it's an optional thing but it's required in Behavior.

What should we do about it? Should we make it optional in `Behavior`? Or should we make it required in `StateMachine`? I think that at the very least one shouldn't be able to `.start` a behavior without this context.

I'm still analyzing how behaviors relate to each other etc so I might have some further thoughts soon.